### DISCUSSION TITLE
Quickest way to check whether the experiments have been finished?

### DISCUSSION THREAD

**Discussion by RuiheLi** [pybamm-team/PyBaMM#2401] [Q&A] [answered]
Dear PyBaMM,

I have been using `try ... except ... else` to run experiments to avoid errors. I wonder what is the quickest way to check whether the experiments have been finished? I try to print out the last voltage, but this cannot deal with Infeasible experiment callback:

`print(sol_new.cycles[-1].steps[-1]["Terminal voltage [V]"].entries[-1])`

Big thanks in advance!

**Comment by valentinsulzer** [comment, marked as answer]
You could define your own callback, see https://github.com/pybamm-team/PyBaMM/blob/develop/examples/notebooks/callbacks.ipynb#L1

Set `self.success = True` at the start and then it `on_experiment_infeasible` and `on_experiment_error` set it to False. Then at the end check if it's True or False
